x-pack/plugins/fleet/server/services/epm/packages/install_state_machine/steps/update_latest_executed_state.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsErrorHelpers } from '@kbn/core/server';
+
+import { PACKAGES_SAVED_OBJECT_TYPE } from '../../../../../constants';
+
+import { auditLoggingService } from '../../../../audit_logging';
+
+import type { InstallContext } from '../_state_machine_package_install';
+
+// Function invoked after each transition
+export const updateLatestExecutedState = async (context: InstallContext) => {
+  const { logger, savedObjectsClient, packageInstallContext, latestExecutedState } = context;
+  const { packageInfo } = packageInstallContext;
+  const { name: pkgName } = packageInfo;
+
+  try {
+    // If the error is of type ConcurrentInstallationError, don't save it in the SO
+    if (latestExecutedState?.error?.includes('Concurrent installation or upgrade')) return;
```

Comment:
@nchaulet I added this return here to avoid saving the error in the SO when it's a `ConcurrentInstallationError`; I didn't have the time to test it properly, I'm going to add tests to cover it.

Reply:
Added https://github.com/elastic/kibana/pull/178657/commits/dc95c4552b0571566ee88258c3ae82ae79102f89#diff-69364ec31cfe1b62b9c4c3fe94270cddcfffdc1d441577945914aeefc28646d6 to cover this specific case.

Reply:
Should we at least log out the error when skipping saving to the SO?